<?xml version="1.0"?>

<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="arduino_actuator_control">

  <material name="black">
    <color rgba="0.0 0.0 0.0 1.0" />
  </material>

  <material name="blue">
    <color rgba="0.0 0.0 0.8 1.0" />
  </material>

  <material name="green">
    <color rgba="0.0 0.8 0.0 1.0" />
  </material>

  <link name="world" />

  <link name="base">
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <box size="0.4 0.4 0.1" />
      </geometry>
    </collision>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <box size="0.4 0.4 0.1" />
      </geometry>
      <material name="blue" />
    </visual>
  </link>

  <link name="rotor">
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <box size="0.4 0.1 0.1" />
      </geometry>
    </collision>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <box size="0.4 0.1 0.1" />
      </geometry>
      <material name="blue" />
    </visual>
  </link>

  <joint name="joint1" type="fixed">
    <parent link="world" />
    <child link="base" />
    <origin xyz="0 0 0" rpy="0 0 0" />
    <axis xyz="0 1 0" />
  </joint>

  <joint name="joint2" type="continuous">
    <parent link="base" />
    <child link="rotor" />
    <origin xyz="0.25 0 0" rpy="0 1.57 0" />
    <axis xyz="0 1 0" />
    <dynamics damping="0.7" />
  </joint>

  <ros2_control name="arduino_actuator_ros2_control" type="system">
    <hardware>
      <plugin>robot_control_system/ArduinoActuatorInterface</plugin>
      <param name="hw_start_duration_sec">0</param>
      <param name="hw_stop_duration_sec">3.0</param>
      <param name="hw_slowdown">100</param>
    </hardware>

    <joint name="joint2">
      <command_interface name="position">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <state_interface name="position" />
    </joint>
  </ros2_control>

</robot>